crates/dbsp/src/circuit/circuit_builder.rs

```diff
@@ -945,6 +949,13 @@ impl GlobalNodeId {
         }
         pid
     }
+
+    pub(crate) fn metrics_id(&self) -> String {
```

Comment:
Forgot to comment on this before. This is an inefficient and somewhat awkward way to generate metric id. Rather than formatting it in one way and then replacing characters, let's just implement the logic to generate it in the right format straight away.